common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -179,12 +179,26 @@ public static StringSearch getStringSearch(
       final UTF8String left,
       final UTF8String right,
       final int collationId) {
+
+    if (collationId == UTF8_BINARY_LCASE_COLLATION_ID) {
+      return getStringSearch(left, right);
+    }
+
     String pattern = right.toString();
     CharacterIterator target = new StringCharacterIterator(left.toString());
     Collator collator = CollationFactory.fetchCollation(collationId).collator;
     return new StringSearch(pattern, target, (RuleBasedCollator) collator);
   }
 
+  private static StringSearch getStringSearch(
```

Comment:
from the method name/signature it isn't apparent that this will lowercase the input strings and then return the string search

Reply:
This is private helper method that is not visible from the outside. Relevant discussion can be found [PR #45704 comment](https://github.com/apache/spark/pull/45704#discussion_r1546513659)

Reply:
@miland-db this function shouldn't lowercase any input, regardless of its access modifier

instead, do this when calling it:
getStringSearch(left.toLowerCase(), right.toLowerCase())
  
  (this goes for other PRs that may use this too)